Documentation/kubernetes-networking.md

```diff
@@ -0,0 +1,91 @@
+
+* External-to-Internal Communication
+    * Accessing services from outside the cluster is generally implemented by configuring external loadbalancers which target all nodes in the cluster. Once traffic arrives at a node, it can be routed to the correct Service backends via the kube-proxy.
+
+See [Kubernetes Networking][kubernetes-network] for more detailed information on the Kubernetes network model and motiviation.
+
+[kubernetes-network]: http://kubernetes.io/v1.0/docs/admin/networking.htm
+
+## Port allocation
+
+The information below describes a minimum set of port allocations used by Kubernetes components. Some of these allocations will be optional depending on the deployment (e.g. if flannel is being used). Additionally, there is likely additional ports a deployer will need to open on their infrastructure (e.g. 22/ssh).
+
+Controller Node Inbound
+
+| Protocol | Port Range | Source                       | Purpose                |
+-----------|------------|------------------------------|------------------------|
+| TCP      | 443        | Worker Nodes & API Consumers | Kubernetes API server. |
+
+Worker Node Inbound
+
+| Protocol | Port Range  | Source                         | Purpose                                                                |
+-----------|-------------|--------------------------------|------------------------------------------------------------------------|
+| TCP      | 10250       | Control Nodes                  | Worker node Kubelet healthcheck port.                                  |
+| TCP      | 30000-32767 | ALL or External Load-balancers | Default port range for [external service][external-service] node ports |
+| TCP      | ALL         | Worker & Control Nodes         | Intra-cluster communication (unnecessary if flannel is used)           |
+| UDP      | 8285        | Worker Nodes                   | flannel overlay network (only required if using flannel)               |
+
+etcd Node Inbound
+
+| Protocol | Port Range | Source        | Purpose                                                  |
+-----------|------------|---------------|----------------------------------------------------------|
+| TCP      | 2379       | Control Nodes | etcd server client API                                   |
+| TCP      | 2379       | Worker Nodes  | etcd server client API (only required if using flannel). |
+| TCP      | 2380       | etcd Nodes    | Used for etcd intra-cluster communication.               |
+
+[external-service]: http://kubernetes.io/v1.0/docs/user-guide/services.html#external-services
+
+## Advanced Configuration
+
+The Coreos Kubernetes documentation will describe either using cloud-provider networking integration (e.g. AWS, GCE), or a software-defined overlay network (e.g. [flannel][coreos-flannel]) to manage the Kubernetes Pod Network. However, in some cases a deployer may want to make use of existing network infrastructure to manage the Kubernetes network themselves.
```

Comment:
The following requirements must be met by your existing infrastructure to use Tectonic with a self-managed network.